Controller and Action detection

therabidbanana opened the pull request:
A possible fix for #10.

therabidbanana committed:
Better controller name detection

crossman commented:
I tested this out on our staging servers and it's a nice improvement. Any chance of getting this merged?

therabidbanana committed:
Security vulnerability in older rpm, force later version

therabidbanana commented:
(Forgot pull requests auto-update - this includes #12 now, but I can split the controller name detection into a separate branch if preferred. )

Xylakant commented:
I'll review and bump the version this week, probably tonight.

crossman commented:
@Xylakant any progress?

abuiles commented on lib/padrino-rpm/instrumentation/padrino.rb:
```diff
@@ -16,14 +16,22 @@ module PadrinoRpm
+          params = found_route.first.params
+          found_route = found_route.first.path.route
+          controller = found_route.controller.to_s
+          short_name = found_route.named.to_s.gsub(/^#{controller}_/, '')
```
I tried to use this, does it actually works for you?  shouldn't it be `found_route.name` ?

abuiles replied:
Ok, there must be something going on with the version, `named` was not longer an `attr_reader` after https://github.com/joshbuddy/http_router/commit/6c7285f9ca1b95a32985596727ab3fb68666d00d#L7L3

abuiles committed:
Fix call to name and some other issues related with http_router.

The attr_reader named was not longer available after the following commit:
   https://github.com/joshbuddy/http_router/commit/6c7285f9ca1b95a32985596727ab3fb68666d00d#L7L3

This changes were probably done against an older version of
http_router.

Also based on
https://github.com/padrino/padrino-framework/commit/587ff7, it seems
like the route has an array of arrays in the first element, handle
that case too.

abuiles committed:
Update newrelic gem dependency.

abuiles committed:
Delete trailing whitespaces.

abuiles committed:
Handle case when compiled_router.recognize doesn't return an HttpRouter.

When requesting an asset compiled_router.recognize returns nil in
the first element

 => found_route = self.class.compiled_router.recognize(request.env)
 => [nil, #<Set: {"GET", "HEAD"}>]

therabidbanana committed:
Merge pull request #1 from abuiles/master

Fix issues with http_router.

therabidbanana commented:
Merged in the latest fixes regarding http_router from @abuiles, but I don't use padrino with New Relic any more so I'm just hoping it works. Anyone still actively using this care to confirm?

abuiles commented:
@therabidbanana I'm using it on production, if someone has a problem, feel free to ping me.

lsdr commented:
Using the @therabidbanana branch in personal projects and in test environments at work and it is all fine. Not yet in production, we don't use gems pulled from Github for production systems.

Which means I'm anxious to see thoses patches merged upstream and a new version of the gem released. :-)

abuiles commented:
@lsdr I've been working nicely for my branch in production, but I guess @Asquera is the man we depend on now :)
